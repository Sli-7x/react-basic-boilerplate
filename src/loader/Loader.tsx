import * as React from 'react';
import { Container, Content, LoaderStyle } from './Loader.style';

export interface ILoaderProps {
  isLoading: boolean;
  full?: boolean;
}

export interface ILoaderState {
  isLoading: boolean;
  time: number;
}

export class Loader extends React.Component<ILoaderProps, ILoaderState> {
  static defaultProps = {
    full: false,
  };

  private timer: any = null;
  private delay: number = 150;

  constructor(props: ILoaderProps) {
    super(props);
    this.state = {
      isLoading: false,
      time: 0,
    };
  }

  componentDidMount() {
    if (this.props.isLoading) {
      if (this.timer) {
        clearTimeout(this.timer);
      }

      this.timer = setTimeout(() => {
        this.setState({ isLoading: true });
      }, this.delay); // tslint:disable-line
    } else {
      clearTimeout(this.timer);
      this.setState({ isLoading: false });
    }
  }

  componentWillUnmount() {
    clearTimeout(this.timer);
    this.setState({ isLoading: false });
  }

  render() {
    if (!this.state.isLoading) {
      return null;
    }
    return (
      <Container full={this.props.full === false ? false : true}>
        <Content>
          <LoaderStyle />
        </Content>
      </Container>
    );
  }
}
