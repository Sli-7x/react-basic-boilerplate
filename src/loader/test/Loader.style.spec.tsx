import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { Container } from '../Loader.style';

describe('Loader.style', () => {
  const props = {
    full: false,
  };

  describe('Container', () => {
    it('should render correctly', () => {
      const tree = renderer.create(<Container {...props} />).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('should cover when passed full=true to Container', () => {
      const tree = renderer.create(<Container full={true} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
