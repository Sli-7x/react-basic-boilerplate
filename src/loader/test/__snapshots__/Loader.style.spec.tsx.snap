// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader.style Container should cover when passed full=true to Container 1`] = `
<div
  className="sc-bdVaJa eQWyBw"
/>
`;

exports[`Loader.style Container should render correctly 1`] = `
<div
  className="sc-bdVaJa bTWqcc"
/>
`;
